src/codegen.cpp

```diff
@@ -9476,7 +9552,7 @@ static jl_llvm_functions_t
     Instruction &prologue_end = ctx.builder.GetInsertBlock()->back();
 
     // step 11a. Emit the entry safepoint
-    if (JL_FEAT_TEST(ctx, safepoint_on_entry))
+    if (params.safepoint_on_entry && JL_FEAT_TEST(ctx, safepoint_on_entry))
```

Comment:
This looks weird, shouldn't ctx be derived from params?

Reply:
ctx derives from params.params not params (I agree the naming choice here was a choice)

Reply:
This broke GPUCompiler.jl's tests, together with https://github.com/JuliaLang/julia/blob/9b63fd91b8ca5ff0c96ee99782a7c1a0fd448987/src/aotcompile.cpp#L797-L798

... which seems like a very ad hoc way of avoiding safepoints? It also introduces an inconsistency wrt. `code_llvm`, which doesn't go through `jl_emit_native`.

Reply:
Yeah, we might just want to emit two separate copies of the function (for inlining and not for inlining) which is what code_llvm essentially does. I went back and forth when I was making the PR with which strategy made more sense.